### PR TITLE
[Fix #5188] Add --no-auto-gen-timestamp option

### LINKED ISSUE #5188: Option for `--auto-gen-config` not to include timestamp when it was generated
## Feature request

As part of our dev cycle we try to manage our `.rubocop_todo.yml` config file actively, with the aim of reducing its size and ultimately eliminating it.

As part of each pull request, we run Rubocop with `--auto-gen-config --no-offense-counts` to check whether we have fortuitously fixed any existing offences.

If we have *not* fixed any offences then `.rubocop_todo.yml` should be unchanged and we can submit the PR for review.

*But* whenever we use `--auto-gen-config` even with `--no-offense-counts` the `.rubocop-todo.yml` file always changes because of the timestamp comment near the top of the file:

```yaml
# on 2017-12-04 17:40:09 +0000 using RuboCop version 0.51.0.
```

We would like the option to generate a `.rubocop_todo.yml` file that does not change unless there are substantive changes to its content.

One solution would be an option (e.g. `--no-timestamp`) that omits this comment from the top of the generated file.

--------

## Expected behavior

```bash
rubocop --auto-gen-config --no-offense-counts --no-timestamp
```

would generate a `.rubocop_todo.yml` file that does not change unless there is a material change to the code it is reviewing (or the version of Rubocop).

## Actual behavior

Rubocop does not have an option to generate a `.rubocop_todo.yml` file that does not change from one run to the next.

## Steps to reproduce the problem

```bash
rubocop --auto-gen-config --no-offense-counts
```

```bash
$ git status
diff --git a/.rubocop_todo.yml b/.rubocop_todo.yml
index 068dc05e6..9c4369644 100644
--- a/.rubocop_todo.yml
+++ b/.rubocop_todo.yml
@@ -1,6 +1,6 @@
 # This configuration was generated by
 # `rubocop --auto-gen-config --exclude-limit 999 --no-offense-counts`
-# on 2017-12-05 17:40:09 +0000 using RuboCop version 0.51.0.
+# on 2017-12-06 09:30:29 +0000 using RuboCop version 0.51.0.
 # The point is for the user to remove these configuration records
 # one by one as the offenses are removed from the code base.
 # Note that changes in the inspected code, or installation of new
```

## RuboCop version

```
$ rubocop -V
0.51.0 (using Parser 2.4.0.2, running on ruby 2.2.2 x86_64-darwin16)
```

### PR DESCRIPTION
**Pull request by dominicsayers**
An option to generate a `.rubocop_todo.yml` file that does not change unless there are substantive changes to its content.

With this option and `--no-offence-counts` combined the `.rubocop_todo.yml` config should not change from run to run unless there are differences in the offences detected. This will allow developers to use automated `git diff` tests to check that the `.rubocop_todo.yml` is up-to-date.

In `options.rb` the verification of the presence of `--auto-gen-config` when any of the three associated options is supplied has been consolidated. This ensures that handling of these parameters is consistent, and as a nice side-effect it allows us to remove the `args` parameter from a chain of methods right up to `define_options`.

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `rake default` or `rake parallel`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/formatter/disabled_config_formatter.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@ class DisabledConfigFormatter < BaseFormatter
       HEADING =
         ['# This configuration was generated by',
          '# `%s`',
-         "# on #{Time.now} using RuboCop version #{Version.version}.",
+         "# %susing RuboCop version #{Version.version}.",
```
Comment (bbatsov) [inline lib/rubocop/formatter/disabled_config_formatter.rb]: New cop idea - string array followed by `.join("\n")` should be replaced with a heredoc. 😄 

/cc @Drenmi @garettarrowood @pocke @koic

**Comment by bbatsov** [comment]
Apart from the failed build, the changes look good to me.

File: lib/rubocop/formatter/disabled_config_formatter.rb (hunk 1 above)
Comment (dominicsayers) [reply lib/rubocop/formatter/disabled_config_formatter.rb]: Fixed in most recent push

**Comment by bbatsov** [comment]
👍

**Pull request merged**